src/utils.ts

```diff
@@ -130,117 +126,97 @@ export function computeDustAmount(currency: ICrypto, txSize: number) {
   return dustAmount;
 }
 
-export function isValidAddress(address: string) {
+/**
+ * Temporarily copied from bitcoinjs-lib master branch (as of 2021-09-02,
+ * commit 7b753caad6a5bf13d40ffb6ae28c2b00f7f5f585) so that we can make use of the
+ * updated bech32 lib version 2.0.0 that supports bech32m. bitcoinjs-lib version 5.2.0
+ * as currently used by wallet-btc uses an older version of bech32 that lacks bech32m support.
+ *
+ * When a new version of bitcoinjs-lib that supports bech32m is released this function can
+ * be removed and calls should be directed to bitcoinjs-lib instead. Our direct dependency
+ * on bech32 lib should also be removed.
+ */
+/* eslint-disable */
+export function fromBech32(address: string): { version: number, prefix: string, data: Buffer} {
+  let result;
+  let version;
+  try {
+    result = bech32.decode(address);
+  } catch (e) {}
+
+  if (result) {
+    version = result.words[0];
+    if (version !== 0) throw new TypeError(address + ' uses wrong encoding');
+  } else {
+    result = bech32m.decode(address);
+    version = result.words[0];
+    if (version === 0) throw new TypeError(address + ' uses wrong encoding');
+  }
+
+  const data = bech32.fromWords(result.words.slice(1));
+
+  return {
+    version,
+    prefix: result.prefix,
+    data: Buffer.from(data),
+  };
+}
+/* eslint-enable */
+
+export function lookupNetwork(networkName: string): bitcoin.Network {
+  if (networkName === 'mainnet') {
+    return bitcoin.networks.bitcoin;
+  }
+  if (networkName === 'testnet') {
+    return bitcoin.networks.testnet;
+  }
+  if (networkName === 'regtest') {
+    return bitcoin.networks.regtest;
+  }
+  throw new TypeError(`Unknown network name ${networkName}`);
+}
+
+export function isValidAddress(address: string, network?: bitcoin.Network) {
+  // We default to mainnet for compatibility reasons. The network parameter is new
+  // and current live-common doesn't pass this argument.
```

Comment:
Does it mean as long as live-common doesn't provide `network`, this will return false for all currencies except Bitcoin?

If so, when network isn't provided, I'd rather fallback to checks that are unspecific to bitcoin currencies (as we have currently), even if it means less checks. Because this is what the current legacy implementation does so we don't really need more, and it would avoid adding the dependency to bitcoinjs-lib in live-common.

But of course if this is compatible with the new bech32m checks? Meaning that checking for a bech32m format doesn't _require_ to know the exact network?

Reply:
Or maybe use a `Currency` param (this live-common can do), and use `cryptoFactory` to get the network from it?

Reply:
> Does it mean as long as live-common doesn't provide network, this will return false for all currencies except Bitcoin?
Oops. My intention was to make a PR to live-common that adds this argument. But I see now that I've misunderstood how this works. I thought live-common:src/family/bitcoin/logic.ts only concerns "bitcoin"/"bitcoin_testnet", but apparently it doesn't. I need to redo this a bit.

> I'd rather fallback to checks that are unspecific to bitcoin currencies
Ok, I suggest the following approach: let isValidAddress take a Currency as an argument instead of a Network and then use the new function cryptoFactory(Currency) to get an ICrypto that will validate the address via a new interface method `ICrypto.validateAddress()`. That way we can have more specific checks for each currency. Would that work? Of course, we also need a fallback (like the one we previously had) for Currencies that don't implement/need special rules.

Reply:
Sounds good! :+1:

Reply:
Ok, back to work....